cosmic-panel-config/src/panel_config.rs

```diff
@@ -402,6 +401,125 @@ impl CosmicPanelConfig {
+        let mut errors = Vec::new();
+
+        match config.get::<String>("name") {
+            Ok(name) => default.name = name,
+            Err(e) => errors.push(e),
+        }
+
+        match config.get::<PanelAnchor>("anchor") {
+            Ok(anchor) => default.anchor = anchor,
+            Err(e) => errors.push(e),
+        }
+
+        match config.get::<bool>("anchor_gap") {
+            Ok(anchor_gap) => default.anchor_gap = anchor_gap,
+            Err(e) => errors.push(e),
+        }
+
+        match config.get::<Layer>("layer") {
+            Ok(layer) => default.layer = layer,
+            Err(e) => errors.push(e),
+        }
+
+        match config.get::<KeyboardInteractivity>("keyboard_interactivity") {
+            Ok(keyboard_interactivity) => default.keyboard_interactivity = keyboard_interactivity,
+            Err(e) => errors.push(e),
+        }
+
+        match config.get::<PanelSize>("size") {
+            Ok(size) => default.size = size,
+            Err(e) => errors.push(e),
+        }
+
+        match config.get::<CosmicPanelOuput>("output") {
+            Ok(output) => default.output = output,
+            Err(e) => errors.push(e),
+        }
+
+        match config.get::<CosmicPanelBackground>("background") {
+            Ok(background) => default.background = background,
+            Err(e) => errors.push(e),
+        }
+
+        match config.get::<Option<(Vec<String>, Vec<String>)>>("plugins_wings") {
+            Ok(plugins_wings) => default.plugins_wings = plugins_wings,
+            Err(e) => errors.push(e),
+        }
+
+        match config.get::<Option<Vec<String>>>("plugins_center") {
+            Ok(plugins_center) => default.plugins_center = plugins_center,
+            Err(e) => errors.push(e),
+        }
+
+        match config.get::<bool>("expand_to_edges") {
+            Ok(expand_to_edges) => default.expand_to_edges = expand_to_edges,
+            Err(e) => errors.push(e),
+        }
+
+        match config.get::<u32>("padding") {
+            Ok(padding) => default.padding = padding,
+            Err(e) => errors.push(e),
+        }
+
+        match config.get::<u32>("spacing") {
+            Ok(spacing) => default.spacing = spacing,
+            Err(e) => errors.push(e),
+        }
+
+        match config.get::<bool>("exclusive_zone") {
+            Ok(exclusive_zone) => default.exclusive_zone = exclusive_zone,
+            Err(e) => errors.push(e),
+        }
+
+        match config.get::<Option<AutoHide>>("autohide") {
+            Ok(autohide) => default.autohide = autohide,
+            Err(e) => errors.push(e),
+        }
+
+        match config.get::<u32>("border_radius") {
+            Ok(border_radius) => default.border_radius = border_radius,
+            Err(e) => errors.push(e),
+        }
+
+        match config.get::<u16>("margin") {
+            Ok(margin) => default.margin = margin,
+            Err(e) => errors.push(e),
+        }
+
+        (default, errors)
+    }
```

Comment:
We should come up with some kind of abstraction in cosmic-config to help with this sort of thing. But this is fine for now.

Reply:
ya :sweat_smile: